app/src/main/java/com/babylon/wallet/android/di/ClientAndApisModule.kt

```diff
@@ -0,0 +1,122 @@
+package com.babylon.wallet.android.di
+
+import com.babylon.wallet.android.data.dapp.PeerdroidClient
+import com.babylon.wallet.android.data.dapp.PeerdroidClientImpl
+import com.babylon.wallet.android.data.gateway.apis.StateApi
+import com.babylon.wallet.android.data.gateway.apis.StreamApi
+import com.babylon.wallet.android.data.gateway.apis.TokenPriceApi
+import com.babylon.wallet.android.data.gateway.apis.TransactionApi
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineDispatcher
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import rdx.works.peerdroid.data.PeerdroidConnector
+import rdx.works.peerdroid.di.IoDispatcher
+import rdx.works.profile.data.model.apppreferences.Radix
+import rdx.works.profile.data.model.currentGateway
+import rdx.works.profile.data.repository.ProfileRepository
+import retrofit2.Converter
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object ClientAndApisModule {
```

Comment:
Maybe change this name to `NetworkApiModule` and rename `NetworkModule` to `HttpModule` ?